src/Features/Core/Portable/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.Analyzer.cs

```diff
@@ -0,0 +1,397 @@
+                        when SupportsTypePattern && CheckTargetExpression(op.ValueOperand) && op.Syntax is TIsExpressionSyntax node:
+                        return new AnalyzedPattern.Type(node);
+
+                    case IIsPatternOperation op
+                        when SupportsSourcePattern && CheckTargetExpression(op.Value) && op.Pattern.Syntax is TPatternSyntax pattern:
+                        return new AnalyzedPattern.Source(pattern);
+
+                    case IParenthesizedOperation op:
+                        return ParsePattern(op.Operand, guards);
+                }
+
+                return null;
+            }
+
+            private enum BoundKind
+            {
+                None,
+                Lower,
+                Higher,
+            }
+
+            private (SyntaxNode Lower, SyntaxNode Higher) GetRangeBounds(IBinaryOperation op)
+            {
+                if (!(op is { LeftOperand: IBinaryOperation left, RightOperand: IBinaryOperation right }))
+                {
+                    return default;
+                }
+
+                return (GetRangeBound(left), GetRangeBound(right)) switch
+                {
+                    ({ Kind: BoundKind.Lower } low, { Kind: BoundKind.Higher } high)
+                        when CheckTargetExpression(low.Expression, high.Expression) => (low.Value.Syntax, high.Value.Syntax),
+                    ({ Kind: BoundKind.Higher } high, { Kind: BoundKind.Lower } low)
+                        when CheckTargetExpression(low.Expression, high.Expression) => (low.Value.Syntax, high.Value.Syntax),
+                    _ => default
+                };
+
+                bool CheckTargetExpression(IOperation left, IOperation right)
+                    => _syntaxFacts.AreEquivalent(left.Syntax, right.Syntax) && this.CheckTargetExpression(left);
+            }
+
+            private static (BoundKind Kind, IOperation Expression, IOperation Value) GetRangeBound(IBinaryOperation op)
+            {
+                return op.OperatorKind switch
+                {
+                    // 5 <= i
+                    LessThanOrEqual when IsConstant(op.LeftOperand) => (BoundKind.Lower, op.RightOperand, op.LeftOperand),
+                    // i <= 5
+                    LessThanOrEqual when IsConstant(op.RightOperand) => (BoundKind.Higher, op.LeftOperand, op.RightOperand),
+                    // 5 >= i
+                    GreaterThanOrEqual when IsConstant(op.LeftOperand) => (BoundKind.Higher, op.RightOperand, op.LeftOperand),
+                    // i >= 5
+                    GreaterThanOrEqual when IsConstant(op.RightOperand) => (BoundKind.Lower, op.LeftOperand, op.RightOperand),
+                    _ => default
+                };
+            }
+
+            private static BinaryOperatorKind Negate(BinaryOperatorKind operatorKind)
+            {
+                return operatorKind switch
+                {
+                    LessThan => GreaterThan,
```

Comment:
needs docs.  i normally thing of `>=` as the negation of `<`

Reply:
perhaps 'flip' makes more sense